src/main/java/org/intellij/sdk/codesync/alerts/PricingAlerts.java

```diff
@@ -34,6 +34,20 @@ public static void setPlanLimitReached(Project project) {
         );
     }
 
+    public static void setPlanLimitReached(Project project, boolean privateRepoCountLimitReached) {
+        // We only want to show notification once every 5 minutes, I have implemented that using locks with an expiry of
+        // 5 minutes. So, skip the notification if lock is not acquired.
+        acquirePricingLock();
```

Comment:
I don't think we need locking here since we want to show notiification only once

Reply:
showPrivateRepoCountReached